crates/sui-adapter-transactional-tests/tests/linters/self_transfer/self_transfer.exp

```diff
@@ -0,0 +1,32 @@
+processed 2 tasks
+
+task 1 'lint'. lines 6-53:
+warning: Instead of transferring object of type `self_transfer::S1` to `tx_context::sender()`, consider returning it from `self_transfer::private_transfer_bad`.
+            This allows a caller to use the object, and enables composability via programmable transactions
+   ┌─ temp_test_file:25:9
+   │
+25 │         transfer::transfer(S1 { id: object::new(ctx), }, tx_context::sender(ctx))
+   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
```

Comment:
We should restructure these messages. 
I would expect something like 
```
warning: Non-composable transfer to sender
   ┌─ temp_test_file:25:9
   │
25 │         transfer::transfer(S1 { id: object::new(ctx), }, tx_context::sender(ctx))
   │         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
            Instead of transferring object of type `self_transfer::S1` to `tx_context::sender`, consider returning it. This allows a caller to use the object, and enables composability via programmable transactions
```

Reply:
This sort of layout will work a bit better (hopefully) once we get these in the IDE